api/api_test.go

```diff
@@ -17,13 +16,15 @@ import (
 	"github.com/gorilla/mux"
 	. "github.com/smartystreets/goconvey/convey"
 
+	authorisation "github.com/ONSdigital/dp-authorisation/v2/authorisation/mock"
 	"github.com/ONSdigital/dp-identity-api/cognito/mock"
 )
 
 func TestSetup(t *testing.T) {
 	Convey("Given an API instance", t, func() {
 		r := mux.NewRouter()
 		ctx := context.Background()
+		authorisationMiddleware := newAuthorisationMiddlwareMock()
```

Comment:
You could just use the `newAuthorisationMiddlwareMock()` constructor in the Setup call and save having to create this variable in the test conveys:

```suggestion
		_, err := Setup(ctx, r, &mock.MockCognitoIdentityProviderClient{}, tt.userPoolId, tt.clientId, tt.clientSecret, tt.clientAuthFlow, tt.allowedDomains, newAuthorisationMiddlwareMock())
```